### PR TITLE
build/pkgs/gambit: Remove

### LINKED ISSUE #30156: Remove broken experimental package gambit
<div id="comment:0"></div>

It is not compatible with python 3.

https://github.com/mkoeppe/sage/runs/876312037

```
  [gambit-15.1.1.p0]   make[2]: Leaving directory '/sage/local/var/tmp/sage/build/gambit-15.1.1.p0/src'
  [gambit-15.1.1.p0]   Traceback (most recent call last):
  [gambit-15.1.1.p0]     File "setup.py", line 32, in <module>
  [gambit-15.1.1.p0]       m.Extension.__dict__ = m._Extension.__dict__
  [gambit-15.1.1.p0]   AttributeError: attribute '__dict__' of 'type' objects is not writable
  [gambit-15.1.1.p0]   Error installing Python API
  [gambit-15.1.1.p0]   
```

Another failure report: `cygwin` linking trouble - https://groups.google.com/g/sage-devel/c/HisDSCo7zlk/m/dAs5K5OrBwAJ

Upstream looks unattended. The last releases are from 2017.

https://github.com/gambitproject/gambit/issues

http://www.gambit-project.org/

https://github.com/gambitproject/gambit/pull/242

https://repology.org/project/gambit-game-theory/versions

Previous tickets: #24840, #24012

CC:  @dimpase @theref @kcrisman @nathanncohen @drvinceknight @tscrim @jhpalmieri

Component: **packages: experimental**

_Issue created by migration from https://trac.sagemath.org/ticket/30156_

### PR DESCRIPTION
**Pull request by mkoeppe**
Removing this broken experimental package, as proposed in 2020.
- Closes #30156

Also removing the module providing outdated "standalone instructions" to use Gambit.

Keeping the various references to Gambit tagged with `# optional - gambit`. This will be updated in the follow-up PR"
- #37809 

The motivation for opening this PR to remove `gambit` now (2024) is that it removes a remaining use of the function `sdh_setup_bdist_wheel` (the old gambit Python package could not be installed using pip). This a step for the modernization of the build frontend for Python packages in the Sage distribution:
- #35618

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by mkoeppe** [comment]
The failures of "CI Linux Incremental" are unrelated and already reported elsewhere.

File: src/sage/game_theory/normal_form_game.py [hunk 1]
```diff
@@ -236,8 +236,7 @@
   `Gambit <http://gambit.sourceforge.net/>`_ [Gambit]_. At present this is
   the only gambit algorithm available in sage but further development will
```
Comment (jhpalmieri) [inline src/sage/game_theory/normal_form_game.py:237]: Is "available in sage" accurate? Are you just leaving it as is for the follow-up PR?

**Comment by jhpalmieri** [comment]
This leaves many references to gambit, but those will presumably be dealt with in #37809. So the PR feels a little incomplete, but (a) it's easier to leave the many references intact for the follow-up, and (b) it's an improvement over the current situation because gambit just doesn't work for us right now. Is that an accurate assessment?

**Comment by mkoeppe** [comment]
Yes, that's the idea.

I could include a link to the follow-up #37809 in the documentation if you think that would be helpful

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**